packages/core/src/common/disposable.ts

```diff
@@ -47,6 +47,29 @@ Object.defineProperty(Disposable, 'NULL', {
     }
 });
 
+/**
+ * Utility for tracking a collection of Disposable objects, specifically for
+ * the sake of finding out when all items in the collection have been disposed.
```

Comment:
Mabe I'm overlooking something. I'm a newbie to these classes, but...

1. The value you mention seems rudimentary. I can just as easily do this
```
const disposables: Disposable[] = [];
// add disposables to array
disposables.forEach(disposable => disposable.dispose());
disposables.length = 0;
```

In other words, with two short lines of code, I can get exactly the functionality you mention. Sure, it's more convenient to just call `dispose()` on a collection object, but surely that convenience is not the key value-add of a class that's 75 lines long. The real magic that I see the utility providing is how it injects itself in the dispose() of its members so that it can find out when all items are disposed and then fire off an event. This lets it be used in other classes to, for example, provide reference counting in a non-intrusive way. I really think that's the special sauce here. But please let me know if you think I'm still off.

2. 
> However, even after calling dispose I can still continue filling the collection with more disposables which may again be disposed.

That's not different than a regular array; see my example above. Oh, and note that the notification mechanism is non-functional once the collection count drops to zero for the first time. So, you really couln't continue using it the way you say and still get the full (and what I'd say is the real) benefit of the utility.

Reply:
I do agree that most things could also be done with an array and maybe I'm off by the real value of the collection. In any case, I think the way you improved the documentation now is way more neutral and highlights all the benefits so that adopters simply can make their own call. Thank you very much for that!